### PR TITLE
Add Odin language

### PR DESCRIPTION
**Pull request by Waqar144**

### CODE REVIEW

**Comment by Waqar144** [comment]
also see https://github.com/compiler-explorer/infra/pull/1474

File: static/modes/odin-mode.ts [hunk 1]
```diff
@@ -0,0 +1,243 @@
+// Copyright (c) 2023, Compiler Explorer Authors
```
Comment (partouf) [inline static/modes/odin-mode.ts:1]: 2024

Comment (Waqar144) [reply static/modes/odin-mode.ts]: Done.

**Comment by partouf** [comment]
Thanks!

I'm applying a couple of fixes after merging

**Comment by partouf** [comment]
https://github.com/compiler-explorer/compiler-explorer/commit/3aff85a216ef4716aef85d583a883869df28be34

**Comment by partouf** [comment]
this is now live

**Comment by Waqar144** [comment]
Thanks a lot for seeing it through! :)

**Pull request merged**